processing/src/main/java/org/apache/druid/query/expression/RandExprMacro.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.expression;
+
+import com.google.common.collect.Iterables;
+import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.math.expr.Expr;
+import org.apache.druid.math.expr.ExprEval;
+import org.apache.druid.math.expr.ExprMacroTable;
+import org.apache.druid.segment.DimensionHandlerUtils;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.DoubleSupplier;
+
+public class RandExprMacro implements ExprMacroTable.ExprMacro
+{
+  @Override
+  public String name()
+  {
+    return "rand";
+  }
+
+  @Override
+  public Expr apply(List<Expr> args)
+  {
+    final DoubleSupplier randomGenerator;
+
+    if (args.isEmpty()) {
+      randomGenerator = () -> ThreadLocalRandom.current().nextDouble();
```

Comment:
If you wanted to make the test in ExpressionsTest.java easier, you could inject an annotated `Supplier<DoubleSupplier>` into this class something like

```
@Inject
RandExprMacro(@ThreadRandom Supplier<DoubleSupplier> localThreadRandomSupplier) {
    ...
}
```

As I'm typing this, it feels like overkill, so feel free to ignore